src/coreclr/debug/createdump/crashinfo.cpp

```diff
@@ -145,7 +150,7 @@ CrashInfo::LogMessage(
 // Gather all the necessary crash dump info.
 //
 bool
-CrashInfo::GatherCrashInfo(MINIDUMP_TYPE minidumpType)
+CrashInfo::GatherCrashInfo(DumpType* dumpType)
```

Comment:
Why is this a pointer now? Is this for the mutability in the call? I am not sure we should be in the business of trying to do this mutability just for that. I think us trying to go out of our way to support minidumps on singlefile is not worth the effort - we should just take the decision on `CreateDump`, print the warning that a dump of type X is getting collected (Or decide to fail it), and pipe the right dump type down.

Reply:
The problem is that only this GatherCrashInfo knows if the dump type is going to change.  We switch to "full" dumps when InitializeDAC() can't find one or it is a Native AOT app.  Without a lot of refactoring there is no way to know that the dump type is going to change outside of GatherCrashInfo.  What we are returning here is the dump type that was "gathered" for the higher level code to display.  Displaying a warning that it changed is a separate change in the code that calls this fucntion.

Reply:
What I am saying is - I don't think we should dynamically change this here. Upfront we should determine if we want to support anything but a full dump, warn about changing to it, and do it upright. In the case of any non-full type failing to initialize the DAC, we should be notified of it. That's a bug.

Reply:
I guess we are discussing what the policy should be instead of the impl details.  Yes, I agree if the DAC initialization fails, createdump should fail and that is what happens in the current code.  The question is what happens if the DAC exists side-by-side for single-file? If it doesn't exists, then it should default to full dumps no matter what the user ask for.  We can print a warning in that case if you want,. This is the only case we change the dump type.  If the DAC exists side-by-side with the single-file app, I think createdump should generate the type of dump the user requested (mini, heap, full, etc).

Reply:
More I think about it, I think I understand what you want as a policy: the user needs to specify mini/heap dump for single-file if they copy a DAC by their app.  And vis-versa, if they specify that they want a mini/heap dump, the DAC is required to be side-by-side. Error out in both cases. Then the DumpType passed into the gather function never changes.

Reply:
Yup, i prefer the latter. Let them explicitly opt in and not have us magically poll assemblies being there. That's a recipe for security issues where it's already an opt in behavior.